sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/processing/flow/FlowSwitch.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2022 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.ui.labmessage.processing.flow;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * Class used for building conditional async flow/chain
+ * 
+ * @param <T>
+ * @param <R>
+ */
+public class FlowSwitch<T, R> {
+
+	private final CompletionStage<ProcessingResult<T>> currentResult;
+	private final CompletionStage<ProcessingResult<R>> switchResult;
+
+	public FlowSwitch(CompletionStage<ProcessingResult<T>> currentResult) {
+		this(currentResult, null);
+	}
+
+	private FlowSwitch(CompletionStage<ProcessingResult<T>> currentResult, CompletionStage<ProcessingResult<R>> switchResult) {
+		this.currentResult = currentResult;
+		this.switchResult = switchResult;
+	}
+
+	public <RR> FlowSwitch<T, RR> when(Function<T, Boolean> condition, SwitchFlow<T, RR> switchFlow) {
```

Comment:
Also here: `SonarLint: Rename this generic name to match the regular expression '^[A-Z][0-9]?$'.`

Furthermore, SonarLint suggests to change `Function<T, Boolean>` to `Predicate<T>`, which seems to be a good suggestion.

`condition.apply(r.getData())` velow would then turn into `condition.test(r.getData())`.